addons/calendar/models/calendar_attendee.py

```diff
@@ -116,12 +116,29 @@ def _send_mail_to_attendees(self, mail_template, force_send=False):
         # get ics file for all meetings
         ics_files = self.mapped('event_id')._get_ics_file()
 
+        # If the mail template has attachments, prepare copies for each attendee (to be added to each attendee's mail)
+        if mail_template.attachment_ids:
+
+            # Setting res_model to ensure attachments are linked to the msg (otherwise only internal users are allowed link attachments)
+            attachments_values = [a.copy_data({'res_id': 0, 'res_model': 'mail.compose.message'})[0] for a in mail_template.attachment_ids]
+            attachments_values *= len(self)
+            attendee_attachment_ids = self.env['ir.attachment'].create(attachments_values).ids
+
+            # Map attendees to their respective attachments
+            template_attachment_count = len(mail_template.attachment_ids)
+            attendee_id_attachment_id_map = {
+                attendee_id: attendee_attachment_ids[i * template_attachment_count: (i + 1) * template_attachment_count]
+                for i, attendee_id in enumerate(self.ids)
```

Comment:
Can't you make it simpler using ``split_every`` and split ``attendee_attachment_ids`` by number of attendees ?